bql/planner/data_access_test.go

```diff
@@ -311,61 +311,65 @@ func TestDataAccessBasicBindings(t *testing.T) {
 		PBinding: "?p",
 		OBinding: "?o",
 	}
+
 	testTable := []struct {
-		t  string
-		sc *table.Cell
-		pc *table.Cell
-		oc *table.Cell
+		t        string
+		sBinding *table.Cell
+		pBinding *table.Cell
+		oBinding *table.Cell
 	}{
 		{
-			t:  n.String() + "\t" + p.String() + "\t" + n.String(),
-			sc: &table.Cell{N: n},
-			pc: &table.Cell{P: p},
-			oc: &table.Cell{N: n},
+			t:        n.String() + "\t" + p.String() + "\t" + n.String(),
+			sBinding: &table.Cell{N: n},
+			pBinding: &table.Cell{P: p},
+			oBinding: &table.Cell{N: n},
 		},
 		{
-			t:  n.String() + "\t" + p.String() + "\t" + p.String(),
-			sc: &table.Cell{N: n},
-			pc: &table.Cell{P: p},
-			oc: &table.Cell{P: p},
+			t:        n.String() + "\t" + p.String() + "\t" + p.String(),
+			sBinding: &table.Cell{N: n},
+			pBinding: &table.Cell{P: p},
+			oBinding: &table.Cell{P: p},
 		},
 		{
-			t:  n.String() + "\t" + p.String() + "\t" + l.String(),
-			sc: &table.Cell{N: n},
-			pc: &table.Cell{P: p},
-			oc: &table.Cell{L: l},
+			t:        n.String() + "\t" + p.String() + "\t" + l.String(),
+			sBinding: &table.Cell{N: n},
+			pBinding: &table.Cell{P: p},
+			oBinding: &table.Cell{L: l},
 		},
 	}
+
 	for _, entry := range testTable {
+		// Setup for test:
 		tpl, err := triple.Parse(entry.t, literal.DefaultBuilder())
 		if err != nil {
-			t.Errorf("triple.Parse failed to parse valid triple %q with error %v", entry.t, err)
+			t.Fatalf(`triple.Parse failed to parse valid triple "%s" with error: %v`, entry.t, err)
 		}
+
+		// Actual test:
 		r, err := tripleToRow(tpl, cls)
 		if err != nil {
-			t.Errorf("tripleToRow for triple %q and clasuse %v, failed with error %v", tpl, cls, err)
-		}
-		if got, want := r["?s"], entry.sc; !reflect.DeepEqual(got, want) {
-			t.Errorf("tripleToRow failed to return right value for binding \"?s\"; got %q, want %q", got, want)
-		}
-		if got, want := r["?p"], entry.pc; !reflect.DeepEqual(got, want) {
-			t.Errorf("tripleToRow failed to return right value for binding \"?p\"; got %q, want %q", got, want)
+			t.Errorf(`tripleToRow for triple "%s" and clause %q failed with error: %v`, tpl, cls, err)
```

Comment:
Please identify the function, input, got and want.
Avoid quoting the string yourself, prefer to use `%q`.
This error message is very verbose. You can convey the same in a more concise, to-the-point way (suggestion below).
And avoid coupling `t.Error` followed by `continue`. In this case, if you can't proceed, please use `t.Failf` instead.
```
t.Fatalf(`tripleToRow(%q, %v) = _, %v; want _, nil`, entry.t, cls, err)
```
I'm not sure the type of `cls` so I suggested `%v`. 
Since you have the first value as string in the test, I suggested to use it instead of the parsed value, since it is probably more readable to the person debugging it.
If both inputs here are too long to be readable when debugging, then prefer adding a description to your test struct, and printing that instead :)

Reply:
This was already changed in a later commit to comply with the standard `tripleToRow(%q) = _, %v; want ...` for tests. I believe you saw this older version here because you were reviewing commit by commit, right?
About the `%q`, I explained in a comment above but I proceeded this way because, after breaking the error to see how it would like for the user, I saw that `%q` prints tabs `\t` as brute `"\t"` strings, which is not good to make the error message more readable when debugging.

Reply:
Sorry I let the other one unnoticed, thanks for updating to `t.Fatalf`.

The way it is now, this is not a case description, but stating the inputs in a verbose way. What I mean by case description is: What path does this exercise? Why did you choose this triple/clause combination?

If the inputs are small enough that the inputs are readable, prefer to do so in a more concise manner like I suggested earlier:

```
t.Fatalf(`tripleToRow("%s", %s) = _, %v; want _, nil`, entry.t, cls, err)
```

Reply:
As we discussed in the Google chat, I decided by the end to just use the standard concise pattern for error messages in tests. Thank you again for the patience and all the replies in the chat Tati. 🙂